src/pytorch_lightning/plugins/io/async_plugin.py

```diff
@@ -0,0 +1,85 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import queue
+from typing import Any, Dict, Optional
+
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.utilities.cloud_io import _ThreadQueue
+from pytorch_lightning.utilities.types import _PATH
+
+
+class AsyncCheckpointIO(CheckpointIO):
+    """CheckpointIO that utilizes :func:`torch.save` and :func:`torch.load` to save and load checkpoints
+    respectively, common for most use cases.
+
+    Args:
+        checkpoint_io: A checkpoint IO plugin that is used as the basis for async checkpointing.
+        interval: Sleep time between each queue check.
+    """
+
+    def __init__(self, checkpoint_io: Optional["CheckpointIO"] = None, interval: float = 2.0) -> None:
+        super().__init__()
+
+        self._checkpoint_io = checkpoint_io
+        self._thread = _ThreadQueue(q=queue.Queue(), interval=interval)
+        self._thread.start()
+
+    @property
```

Comment:
How about moving this property to base `CheckpointIO`? The advantage would be to have the setter recursive and supporting multiple wrappers around each other such as `SplitCheckpointIO(AsyncCheckpointIO(HPUCheckpointIO()))`

Reply:
I don't think this fits in the base class. Perhaps in a middle step between the base and this one. `_WrappingCheckpointIO`? Then `is_wrapper` would also be an `isinstance` check on this middle class